File: HolyDay/Models/SupporterTier.swift
//
//  SupporterTier.swift
//  HolyDay
//
//  Created by Matthias Cadet on 14/05/2026.
//

import SwiftUI

enum SupporterTier: Int {
  case ami = 0
  case genereux = 1
  case bienfaiteur = 2

  var title: String {
    switch self {
    case .ami: return String(localized: "tier.ami")
    case .genereux: return String(localized: "tier.genereux")
    case .bienfaiteur: return String(localized: "tier.bienfaiteur")
    }
  }

  var emoji: String {
    switch self {
    case .ami: return "☕"
    case .genereux: return "🙏"
    case .bienfaiteur: return "✨"
    }
  }

  var tipLabel: LocalizedStringKey {
    switch self {
    case .ami: return "tip.tier.0.label"
    case .genereux: return "tip.tier.1.label"
    case .bienfaiteur: return "tip.tier.2.label"
    }
  }

  var phrase: LocalizedStringKey {
    switch self {
    case .ami: return "paywall.tip.ami.phrase"
    case .genereux: return "paywall.tip.genereux.phrase"
    case .bienfaiteur: return "paywall.tip.bienfaiteur.phrase"
    }
  }

  var icon: String {
    switch self {
    case .ami: return "heart.fill"
    case .genereux: return "star.fill"
    case .bienfaiteur: return "crown.fill"
    }
  }

  var color: Color {
    switch self {
    case .ami: return AppTheme.thanksgivingGold
    case .genereux: return AppTheme.confessionBlue
    case .bienfaiteur: return AppTheme.adorationPurple
    }
  }

  static func tier(for productIdentifier: String) -> SupporterTier? {
    if productIdentifier.contains("tip_large") { return .bienfaiteur }
    if productIdentifier.contains("tip_medium") { return .genereux }
    if productIdentifier.contains("tip_small") { return .ami }
    return nil
  }
}

File: android/app/src/main/java/com/matthiascadet/holyday/data/model/SupporterTier.kt
package com.matthiascadet.holyday.data.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Star
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.matthiascadet.holyday.R
import com.matthiascadet.holyday.ui.theme.AppTheme

enum class SupporterTier(
    val rank: Int,
    val titleRes: Int,
    val badgeNameRes: Int,
    val emoji: String,
    val phraseRes: Int,
    val icon: ImageVector,
) {
    AMI(0, R.string.tier_ami, R.string.badge_name_ami, "❤️", R.string.paywall_tip_ami_phrase, Icons.Filled.Favorite),
    GENEREUX(1, R.string.tier_genereux, R.string.badge_name_genereux, "⭐️", R.string.paywall_tip_genereux_phrase, Icons.Filled.Star),
    BIENFAITEUR(2, R.string.tier_bienfaiteur, R.string.badge_name_bienfaiteur, "✨", R.string.paywall_tip_bienfaiteur_phrase, Icons.Filled.AutoAwesome);

    @Composable
    fun color(): Color = when (this) {
        AMI -> AppTheme.colors.thanksgivingGold
        GENEREUX -> AppTheme.colors.confessionBlue
        BIENFAITEUR -> AppTheme.colors.adorationPurple
    }

    companion object {
        fun forProductIdentifier(productIdentifier: String): SupporterTier? = when {
            productIdentifier.contains("tip_large") -> BIENFAITEUR
            productIdentifier.contains("tip_medium") -> GENEREUX
            productIdentifier.contains("tip_small") -> AMI
            else -> null
        }
    }
}
